### ISSUE TITLE
Build fails with cmake

### ISSUE THREAD

**Issue by Jacalz** [rncbc/qtractor#232]
Building with autotools works just as expected, but building with cmake errors and fails when it can't find the headers for `aeffectx.h` and `dssi.h`. I don't have any dssi libraries installed so it should idealy see that they aren't there and then turn off dssi support like the autotools build does. Regarding `aeffectx.h`, I don't know where that's from but I suspect that I don't install that either and that the cmake build just fails because it's set to yes by default and doesn't handle a no correctly.

## Summary:
- VST plug-in support is broken in Cmake build system.
- Cmake build system exists when it can't find optional libraries that are enabled by default (DSSI for example).
- Autotools build won't find libmad for some reason, while Cmake does.

**Comment by rncbc** [comment]
cmake is still not the official build-system; and it will remain like so for quite some time.
thanks anyway for trying it
cheers

**Comment by Jacalz** [comment]
Yeah, but you could at least try to fix the issues with it?

**Comment by Jacalz** [comment]
I mean it's a bit weird that the autotools build seems to advertise dssi support as disabled and vst support as on while cmake reports dssi as enabled and vst support dissabled.

Autotools:
![image](https://user-images.githubusercontent.com/25466657/66701138-344d3e00-ecf9-11e9-9650-68c3d0a95116.png)

Cmake:
![image](https://user-images.githubusercontent.com/25466657/66701165-a1f96a00-ecf9-11e9-8b00-542848de79e1.png)

No packages have changed between builds. So it basically looks like VST support is broken in the cmake build system. Italso shouldn't error when it can't find an optional library, that should only be done if it's required. It should just say that it can't find it and advertise it ass off above.

**Comment by rncbc** [comment]
>  autotools build system seems to not find libmad for some reason. It is actually installed.

check config.log for reasons why it doesn't ...
they're both supposed to check on libmad via pkg-config though
also for dssi.h in the same manner
it seems something is not working in your system environment in particular and not quite the difference between autotools and cmake configurations... 
can you tell what system is that of yours and where exactly libmad-dev(el) and/or dssi-dev(el) are or were installed in the first place?
seeya

**Comment by Jacalz** [comment]
I am building inside `solbuild` it is a sandboxed build environment fro the Solus linux distrubution. The build environment works well with pretty much every software, I don't know whats going on here. I don't really know how to reach the config.log file in the build environment. 

DSSI is not in the repo and will thus not be able to install in to the build environment. Libmad-devel is installed to `/usr/lib64/libmad.so` and `/usr/include/mad.h`.

**Comment by rncbc** [comment]
- the config.log file is produced whenever you run "`./configure`" and should be found in the same directory ("`./config.log`");
- for DSSI support  is mostly just the "`dssi.h`" header file that is checked for its existence on both tools--can't figure out why it's positive on one and not the other;
- libmad should also install "`/usr/lib64/pkgconfig/mad.pc`" which in fact is the _de jure_ source for checking for the development package installation on both sides--again why onne picks it up alright and not the other puzzles me. sorry.

**Comment by hasebastian** [comment]
The check_include_file macro is not used properly. It does not accept an already existing variable, i believe. Thus, checks like

```
if (CONFIG_LADSPA)
  check_include_file (ladspa.h CONFIG_LADSPA)
endif ()
```

should read

```
if (CONFIG_LADSPA)
  check_include_file (ladspa.h HAVE_LADSPA_H)
  if (NOT HAVE_LADSPA_H)
    set (CONFIG_LADSPA 0)
  endif ()
endif ()
```

**Comment by rncbc** [comment]
changes were made to the top CMakeLists.txt file, in a variation to yours advice
please check 70c3676e
thanks

**Comment by hasebastian** [comment]
That does not work. When `check_include_file` does not find the given file, e.g. ladspa.h, the associated variable `HAVE_LADSPA_H` does not contain false, it simply does not exist and the call to `set(CONFIG_LADSPA HAVE_LADSPA_H)` has no effect. Thus, `CONFIG_LADSPA` is still `ON`. I guess.

There are more things that need attention, i can make a pull request when i'm done.

**Comment by rncbc** [comment]
maybe this is better 1d20e3f0 :)

thanks again

**Comment by Jacalz** [comment]
I get the following errors when building using cmake with the latest commit:
```make
In file included from /usr/include/gtk-2.0/gtk/gtkobject.h:37,
                 from /usr/include/gtk-2.0/gtk/gtkwidget.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkcontainer.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkbin.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkwindow.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkdialog.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkaboutdialog.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:33,
                 from /home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorLv2Plugin.cpp:1367:
/usr/include/gtk-2.0/gtk/gtktypeutils.h:236:64: warning: ‘GTypeDebugFlags’ is deprecated [-Wdeprecated-declarations]
  236 | void            gtk_type_init   (GTypeDebugFlags    debug_flags);
      |                                                                ^
In file included from /usr/include/glib-2.0/gobject/gobject.h:24,
                 from /usr/include/glib-2.0/gobject/gbinding.h:29,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from /home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorLv2Plugin.cpp:1367:
/usr/include/glib-2.0/gobject/gtype.h:685:3: note: declared here
  685 | } GTypeDebugFlags GLIB_DEPRECATED_TYPE_IN_2_36;
      |   ^~~~~~~~~~~~~~~
In file included from /usr/include/gtk-2.0/gtk/gtk.h:89,
                 from /home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorLv2Plugin.cpp:1367:
/usr/include/gtk-2.0/gtk/gtkfilechooserbutton.h:59:8: warning: unnecessary parentheses in declaration of ‘__gtk_reserved1’ [-Wparentheses]
   59 |   void (*__gtk_reserved1);
      |        ^
/usr/include/gtk-2.0/gtk/gtkfilechooserbutton.h:60:8: warning: unnecessary parentheses in declaration of ‘__gtk_reserved2’ [-Wparentheses]
   60 |   void (*__gtk_reserved2);
      |        ^
/usr/include/gtk-2.0/gtk/gtkfilechooserbutton.h:61:8: warning: unnecessary parentheses in declaration of ‘__gtk_reserved3’ [-Wparentheses]
   61 |   void (*__gtk_reserved3);
      |        ^
/usr/include/gtk-2.0/gtk/gtkfilechooserbutton.h:62:8: warning: unnecessary parentheses in declaration of ‘__gtk_reserved4’ [-Wparentheses]
   62 |   void (*__gtk_reserved4);
      |        ^
/usr/include/gtk-2.0/gtk/gtkfilechooserbutton.h:63:8: warning: unnecessary parentheses in declaration of ‘__gtk_reserved5’ [-Wparentheses]
   63 |   void (*__gtk_reserved5);
      |        ^
/usr/include/gtk-2.0/gtk/gtkfilechooserbutton.h:64:8: warning: unnecessary parentheses in declaration of ‘__gtk_reserved6’ [-Wparentheses]
   64 |   void (*__gtk_reserved6);
      |        ^
/usr/include/gtk-2.0/gtk/gtkfilechooserbutton.h:65:8: warning: unnecessary parentheses in declaration of ‘__gtk_reserved7’ [-Wparentheses]
   65 |   void (*__gtk_reserved7);
      |        ^
In file included from /usr/include/gtk-2.0/gtk/gtktoolitem.h:31,
                 from /usr/include/gtk-2.0/gtk/gtktoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtkmenutoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtk.h:126,
                 from /home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorLv2Plugin.cpp:1367:
/usr/include/gtk-2.0/gtk/gtktooltips.h:73:12: warning: ‘GTimeVal’ is deprecated: Use 'GDateTime' instead [-Wdeprecated-declarations]
   73 |   GTimeVal last_popdown;
      |            ^~~~~~~~~~~~
In file included from /usr/include/glib-2.0/glib/galloca.h:32,
                 from /usr/include/glib-2.0/glib.h:30,
                 from /usr/include/glib-2.0/gobject/gbinding.h:28,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from /home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorLv2Plugin.cpp:1367:
/usr/include/glib-2.0/glib/gtypes.h:549:26: note: declared here
  549 | typedef struct _GTimeVal GTimeVal GLIB_DEPRECATED_TYPE_IN_2_62_FOR(GDateTime);
      |                          ^~~~~~~~
In file included from /usr/include/gtk-2.0/gtk/gtk.h:173,
                 from /home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorLv2Plugin.cpp:1367:
/usr/include/gtk-2.0/gtk/gtkstatusicon.h:76:8: warning: unnecessary parentheses in declaration of ‘__gtk_reserved1’ [-Wparentheses]
   76 |   void (*__gtk_reserved1);
      |        ^
/usr/include/gtk-2.0/gtk/gtkstatusicon.h:77:8: warning: unnecessary parentheses in declaration of ‘__gtk_reserved2’ [-Wparentheses]
   77 |   void (*__gtk_reserved2);
      |        ^
In file included from /home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorLv2Plugin.h:27,
                 from /home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorLv2Plugin.cpp:26:
/home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorLv2Plugin.cpp: In member function ‘virtual bool qtractorLv2PluginType::open()’:
/usr/include/lilv-0/lilv/lilv.h:355:17: warning: unnecessary parentheses in declaration of ‘iter’ [-Wparentheses]
  355 |  for (LilvIter* (iter) = lilv_ ## colltype ## _begin(collection); \
      |                 ^
/home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorLv2Plugin.cpp:1529:3: note: in expansion of macro ‘LILV_FOREACH’
 1529 |   LILV_FOREACH(nodes, iter, nodes) {
      |   ^~~~~~~~~~~~
/usr/include/lilv-0/lilv/lilv.h:355:17: warning: unnecessary parentheses in declaration of ‘iter’ [-Wparentheses]
  355 |  for (LilvIter* (iter) = lilv_ ## colltype ## _begin(collection); \
      |                 ^
/home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorLv2Plugin.cpp:1544:3: note: in expansion of macro ‘LILV_FOREACH’
 1544 |   LILV_FOREACH(uis, iter, uis) {
      |   ^~~~~~~~~~~~
/home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorLv2Plugin.cpp: In static member function ‘static QStringList qtractorLv2PluginType::lv2_plugins()’:
/usr/include/lilv-0/lilv/lilv.h:355:17: warning: unnecessary parentheses in declaration of ‘iter’ [-Wparentheses]
  355 |  for (LilvIter* (iter) = lilv_ ## colltype ## _begin(collection); \
      |                 ^
/home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorLv2Plugin.cpp:2025:3: note: in expansion of macro ‘LILV_FOREACH’
 2025 |   LILV_FOREACH(plugins, iter, g_lv2_plugins) {
      |   ^~~~~~~~~~~~
/home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorLv2Plugin.cpp: In constructor ‘qtractorLv2Plugin::qtractorLv2Plugin(qtractorPluginList*, qtractorLv2PluginType*)’:
/usr/include/lilv-0/lilv/lilv.h:355:17: warning: unnecessary parentheses in declaration of ‘iter’ [-Wparentheses]
  355 |  for (LilvIter* (iter) = lilv_ ## colltype ## _begin(collection); \
      |                 ^
/home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorLv2Plugin.cpp:2496:4: note: in expansion of macro ‘LILV_FOREACH’
 2496 |    LILV_FOREACH(nodes, iter, presets) {
      |    ^~~~~~~~~~~~
/home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorLv2Plugin.cpp: In member function ‘virtual void qtractorLv2Plugin::openEditor(QWidget*)’:
/usr/include/lilv-0/lilv/lilv.h:355:17: warning: unnecessary parentheses in declaration of ‘iter’ [-Wparentheses]
  355 |  for (LilvIter* (iter) = lilv_ ## colltype ## _begin(collection); \
      |                 ^
/home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorLv2Plugin.cpp:3115:2: note: in expansion of macro ‘LILV_FOREACH’
 3115 |  LILV_FOREACH(uis, iter, m_lv2_uis) {
      |  ^~~~~~~~~~~~
/home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorLv2Plugin.cpp: In member function ‘virtual void qtractorLv2Plugin::idleEditor()’:
/home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorLv2Plugin.cpp:3533:4: error: ‘qtractorPluginParamValuesCommand’ was not declared in this scope; did you mean ‘qtractorPluginParamDisplay’?
 3533 |    qtractorPluginParamValuesCommand *pParamValuesCommand
      |    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |    qtractorPluginParamDisplay
/home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorLv2Plugin.cpp:3533:38: error: ‘pParamValuesCommand’ was not declared in this scope
 3533 |    qtractorPluginParamValuesCommand *pParamValuesCommand
      |                                      ^~~~~~~~~~~~~~~~~~~
/home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorLv2Plugin.cpp:3534:11: error: expected type-specifier before ‘qtractorPluginParamValuesCommand’
 3534 |     = new qtractorPluginParamValuesCommand(
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorLv2Plugin.cpp:3548:12: error: type ‘<type error>’ argument given to ‘delete’, expected pointer
 3548 |     delete pParamValuesCommand;
      |            ^~~~~~~~~~~~~~~~~~~
In file included from /home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorLv2Plugin.h:27,
                 from /home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorLv2Plugin.cpp:26:
/home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorLv2Plugin.cpp: In member function ‘void qtractorLv2Plugin::lv2_patch_properties(const char*)’:
/usr/include/lilv-0/lilv/lilv.h:355:17: warning: unnecessary parentheses in declaration of ‘iter’ [-Wparentheses]
  355 |  for (LilvIter* (iter) = lilv_ ## colltype ## _begin(collection); \
      |                 ^
/home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorLv2Plugin.cpp:4178:2: note: in expansion of macro ‘LILV_FOREACH’
 4178 |  LILV_FOREACH(nodes, iter, properties) {
      |  ^~~~~~~~~~~~
/home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorLv2Plugin.cpp: In constructor ‘qtractorLv2PluginParam::qtractorLv2PluginParam(qtractorLv2Plugin*, long unsigned int)’:
/usr/include/lilv-0/lilv/lilv.h:355:17: warning: unnecessary parentheses in declaration of ‘iter’ [-Wparentheses]
  355 |  for (LilvIter* (iter) = lilv_ ## colltype ## _begin(collection); \
      |                 ^
/home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorLv2Plugin.cpp:5126:3: note: in expansion of macro ‘LILV_FOREACH’
 5126 |   LILV_FOREACH(scale_points, iter, points) {
      |   ^~~~~~~~~~~~
make[3]: *** [src/CMakeFiles/qtractor.dir/build.make:2997: src/CMakeFiles/qtractor.dir/qtractorLv2Plugin.cpp.o] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: Leaving directory '/home/build/YPKG/root/qtractor/build/qtractor.git'
make[2]: *** [CMakeFiles/Makefile2:98: src/CMakeFiles/qtractor.dir/all] Error 2
make[2]: Leaving directory '/home/build/YPKG/root/qtractor/build/qtractor.git'
make[1]: *** [Makefile:130: all] Error 2
make[1]: Leaving directory '/home/build/YPKG/root/qtractor/build/qtractor.git/src'
make: *** [Makefile:48: sub-src-make_first] Error 2
```
</details>

**Comment by rncbc** [comment]
@Jacalz : you or the (still faulty?) cmake build is probably skipping LV2 Programs support ... please check that out on the cmake output (re. check_include_file lv2_programs.h).

anyway, fixed the error upstream in 49527e5e but the above is still worth checking still.

**Comment by Jacalz** [comment]
It failed on finding vestige.h file now.
```make
g++ -c -pipe -mtune=generic -march=x86-64 -g2 -O2 -pipe -fPIC -D_FORTIFY_SOURCE=2 -fstack-protector-strong --param ssp-buffer-size=32 -fasynchronous-unwind-tables -ftree-vectorize -feliminate-unused-debug-types -Wall -Wno-error -Wp,-D_REENTRANT -Wall -W -D_REENTRANT -fPIC -DQT_NO_DEBUG -DQT_CORE_LIB -I. -I /usr/include/QtCore -I. -I/usr/lib64/qt5/mkspecs/linux-g++ -o qtractor_plugin_scan.o qtractor_plugin_scan.cpp
qtractor_plugin_scan.cpp:499:10: fatal error: vestige.h: No such file or directory
  499 | #include <vestige.h>
      |          ^~~~~~~~~~~
compilation terminated.
make[1]: *** [Makefile.qtractor_plugin_scan:429: qtractor_plugin_scan.o] Error 1
make[1]: Leaving directory '/home/build/YPKG/root/qtractor/build/qtractor.git/src'
make: *** [Makefile:73: sub-src-qtractor_plugin_scan-pro-make_first] Error 2
Command '['/bin/bash', '--norc', '--noprofile', '/tmp/ypkg-buildGsC_B3']' returned non-zero exit status 2
[Build] build failed
```

**Comment by Jacalz** [comment]
I reverted the vestige changes in the latest commit (see https://github.com/Jacalz/qtractor/commit/21cf165f12fd55c186bef314711e571bb425a4b7) and I managed to get to the very end of the build before it failed on libmad:
```make
[100%] Linking CXX executable qtractor
/usr/bin/ld: CMakeFiles/qtractor.dir/qtractorAudioMadFile.cpp.o: in function `qtractorAudioMadFile::close()':
/home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorAudioMadFile.cpp:470: undefined reference to `mad_frame_finish'
/usr/bin/ld: /home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorAudioMadFile.cpp:471: undefined reference to `mad_stream_finish'
/usr/bin/ld: CMakeFiles/qtractor.dir/qtractorAudioMadFile.cpp.o: in function `qtractorAudioMadFile::input()':
/home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorAudioMadFile.cpp:220: undefined reference to `mad_stream_buffer'
/usr/bin/ld: CMakeFiles/qtractor.dir/qtractorAudioMadFile.cpp.o: in function `qtractorAudioMadFile::decode()':
/home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorAudioMadFile.cpp:242: undefined reference to `mad_frame_decode'
/usr/bin/ld: /home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorAudioMadFile.cpp:263: undefined reference to `mad_synth_frame'
/usr/bin/ld: CMakeFiles/qtractor.dir/qtractorAudioMadFile.cpp.o: in function `qtractorAudioMadFile::open(QString const&, int)':
/home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorAudioMadFile.cpp:106: undefined reference to `mad_stream_init'
/usr/bin/ld: /home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorAudioMadFile.cpp:107: undefined reference to `mad_frame_init'
/usr/bin/ld: /home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorAudioMadFile.cpp:108: undefined reference to `mad_synth_init'
/usr/bin/ld: /home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorAudioMadFile.cpp:125: undefined reference to `mad_header_decode'
/usr/bin/ld: CMakeFiles/qtractor.dir/qtractorAudioMadFile.cpp.o: in function `qtractorAudioMadFile::seek(unsigned long)':
/home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorAudioMadFile.cpp:421: undefined reference to `mad_frame_finish'
/usr/bin/ld: /home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorAudioMadFile.cpp:422: undefined reference to `mad_stream_finish'
/usr/bin/ld: /home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorAudioMadFile.cpp:424: undefined reference to `mad_stream_init'
/usr/bin/ld: /home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorAudioMadFile.cpp:425: undefined reference to `mad_frame_init'
/usr/bin/ld: /home/build/YPKG/root/qtractor/build/qtractor.git/src/qtractorAudioMadFile.cpp:426: undefined reference to `mad_synth_init'
collect2: error: ld returned 1 exit status
```

**Comment by hasebastian** [comment]
The detection of absence of dependencies that are to be found using `pkg_check_modules` (libmad, libsamplerate, librubberband,..) is flawed.

For example, the libmad dependency detection could look like this:

```
if (CONFIG_LIBMAD)
  pkg_check_modules (MAD mad)
  if (MAD_FOUND)
    include_directories (${MAD_INCLUDE_DIRS})
    link_directories (${MAD_LIBRARY_DIRS})
    link_libraries (${MAD_LIBRARIES})
  else ()
    message (WARNING "*** MAD library not found.")
    set(CONFIG_LIBMAD 0)
  endif ()
endif ()
```

**Comment by rncbc** [comment]
yeah right, this whole cmake thing is/was deeply flawed to start with :)
i can't even think why it's being posed as an option to replace good old and reliable autotools/qmake, lol.
anyway, I'll try to keep and follow your kind advice once more and indulge... but, trust me, i'm not getting in anyway sympathetic to this crap--for the better and worse, autotools will stay the supported and official build configuration tool for yet many and then some more years (at least until the Qt-company drops qmake for good ;)). 
cheers

**Comment by Jacalz** [comment]
I put down some time in to trying to fix this and I ultimately failed over the course of four or five attempts. The feeling that I have got is that cmake is just good and fast for people building the software, not those who write it. Meson seems to be the future. I might try to make one for one o your softwares just to learn the thing.

**Comment by rncbc** [comment]
@hasebastian : follow up in  28e37222 . thanks

**Comment by hasebastian** [comment]
28e3722 looks good. I did a quick check with most of the dependencies uninstalled. And it works as expected, it builds and runs fine.

**Comment by Jacalz** [comment]
I had to do the changes in https://github.com/rncbc/qtractor/pull/235 to make it build for me but there is still one issue that exists for me.

The autotools build show these in the about menu: (ignore the missing libmad, i fixed that today)
![image](https://user-images.githubusercontent.com/25466657/68991937-aa971000-0864-11ea-81f0-803bc625ac61.png)

The working build using https://github.com/rncbc/qtractor/pull/235 shows these:
![image](https://user-images.githubusercontent.com/25466657/68991965-f053d880-0864-11ea-9c6e-ecd0a5ab3637.png)

In other words, the cmake build system marks vst support as mussing while the autotools build system marks it as included.

**Comment by Jacalz** [comment]
All issues that I have encountered are all fixed as of https://github.com/rncbc/qtractor/commit/36c53e3c7da1c64bdacaf480c1481fd3f2f9834a. Thank you both for helping out getting the build system up to a great standard. I will make sure to be using the cmake build system with the ninja back-end in the solus package, when the next release is ready.
